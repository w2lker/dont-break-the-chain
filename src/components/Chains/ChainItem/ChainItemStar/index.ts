import ChainItemStar from "./ChainItemStar";
export default ChainItemStar;