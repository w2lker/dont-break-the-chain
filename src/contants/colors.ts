import colorBlue from '@material-ui/core/colors/blue';
import colorBrown from '@material-ui/core/colors/brown';
import colorCyan from '@material-ui/core/colors/cyan';
import colorDeepPurple from '@material-ui/core/colors/deepPurple';
import colorGreen from '@material-ui/core/colors/green';
import colorIndigo from '@material-ui/core/colors/indigo';
import colorOrange from '@material-ui/core/colors/orange';
import colorPink from '@material-ui/core/colors/pink';
import colorPurple from '@material-ui/core/colors/purple';
import colorTeal from '@material-ui/core/colors/teal';

export type colorsKeys = 'pink' | 'purple' | 'deepPurple' | 'indigo' | 'blue' | 'cyan' | 'teal' | 'green' | 'orange' | 'brown';

export const colors: {[color: string]: any}  = {
  blue: colorBlue,
  brown: colorBrown,
  pink: colorPink,
  cyan: colorCyan,
  deepPurple: colorDeepPurple,
  indigo: colorIndigo,
  green: colorGreen,
  orange: colorOrange,
  purple: colorPurple,
  teal: colorTeal,
};
