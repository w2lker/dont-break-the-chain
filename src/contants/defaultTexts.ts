const defaultTexts = {
  navigationMenu: {
    chains: 'Chains',
    habits: 'Habits',
    profile: 'Profile',
  },
  habits: {
    header: {
      title: 'My Habits',
    },
  },
  profile: {
    header: {
      title: 'Profile',
    },
    stats: {
      total: {
        description: 'points received',
      },
      days: {
        subtitle: 'days',
        description: 'longest chain',
      },
    },
  },
};

export default defaultTexts;
