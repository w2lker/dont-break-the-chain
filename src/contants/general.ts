import { RequestStatuses } from '../models/general';

export const requestStatuses = {
  active: 'active',
  success: 'success',
  error: 'error',
};
