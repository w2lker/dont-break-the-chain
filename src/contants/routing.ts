export const routing = {
  default: '/',
  chains: '/chains',
  habits: '/habit',
  profile: '/account',
};
