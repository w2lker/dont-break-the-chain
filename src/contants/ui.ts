export const defaultIcons = {
  navigationMenu: {
    chains: 'layers',
    habits: 'flag-checkered',
    profile: 'account',
  },
};

// TODO: provide real placeholder image and attach link to it here
export const avatarPlaceholder = './placeholder.png';
