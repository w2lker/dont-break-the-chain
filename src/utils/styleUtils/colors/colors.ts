import {bonWarning} from '../helpers/warning/warning';

export interface IRgbColor {
  red: number;
  green: number;
  blue: number;
}

const black: IRgbColor = {
  red: 0,
  green: 0,
  blue: 0,
};

function regularHexToRgb(hex: string): IRgbColor {
  const [red, green, blue] = [0, 2, 4]
    .map(offset => hex.substring(offset, offset + 2))
    .map(hexCh => parseInt(hexCh, 16));
  return { red, green, blue };
}

export function hexToRgbMapped(hex?: string): IRgbColor {
  const warnType = 'HEX to RGB converter';
  if (!hex) {
    return black;
  }
  if (hex.length === 1) {
    return regularHexToRgb(hex.repeat(6));
  }

  if (hex.length === 2) {
    return regularHexToRgb(hex.repeat(3));
  }

  if (hex.length === 3) {
    const [r, g, b] = hex.split('') as string[];
    return regularHexToRgb(`${r}${r}${g}${g}${b}${b}`);
  }

  if (hex.length === 4 || hex.length === 5) {
    const hex3 = hex.slice(0, 3);
    // tslint:disable-next-line:no-console
    bonWarning(`can't verify converting from HEX to RGB — received HEX value with ${hex.length} digits. It is parsed as ${hex3} value`, warnType);
    return hexToRgbMapped(hex3);
  }

  if (hex.length > 6) {
    const hex6 = hex.slice(0, 6);
    // tslint:disable-next-line:no-console
    bonWarning(`can't verify converting from HEX to RGB — received HEX value with ${hex.length} digits. It is parsed as ${hex6} value`, warnType);
    return regularHexToRgb(hex6);
  }

  return regularHexToRgb(hex);
}
