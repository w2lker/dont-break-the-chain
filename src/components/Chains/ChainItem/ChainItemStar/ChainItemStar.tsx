import React from 'react';

import MaterialDesignIcon from '../../../basic/Material-Icon';

import { colors, colorsKeys } from '../../../../contants/colors';

export interface IChainItemStarProps {
  value: number;
  color: colorsKeys;
}

const ChainItemStar: React.FC<IChainItemStarProps> = (props) => {
  const { value } = props;
  if (!value || value < 4) {
    return null;
  }

  const color = colors[props.color || 'blue'][500];

  return (
    <MaterialDesignIcon
      name="star"
      style={{ color }}
    />
  );
};

export default ChainItemStar;
