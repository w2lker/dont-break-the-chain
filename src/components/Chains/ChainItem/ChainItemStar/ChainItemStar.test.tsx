import { mount, shallow } from 'enzyme';
import React from 'react';

import ChainItemStar, { IChainItemStarProps } from './ChainItemStar';

import { colors } from '../../../../contants/colors';

describe('Chain Item Score', () => {
  const emptyProps: IChainItemStarProps = {
    // @ts-ignore
    color: null,
    // @ts-ignore
    value: null,
  };

  const sampleProps: IChainItemStarProps = {
    color: 'green',
    value: 6,
  };

  it('matches snapshot', () => {
    const component = mount(<ChainItemStar {...sampleProps} />);
    expect(component.debug()).toMatchSnapshot();
  });

  it('skips render without value', () => {
    const newProps: IChainItemStarProps = {
      ...sampleProps,
      // @ts-ignore
      value: null,
    };
    const component = shallow(<ChainItemStar {...newProps} />);
    expect(component.isEmptyRender()).toBe(true);
  });

  it('renders with default color', () => {
    const newProps: IChainItemStarProps = {
      ...sampleProps,
        // @ts-ignore
      color: null,
    };
    const sampleStyle = {
      color: colors.blue[500],
    };
    const component = shallow(<ChainItemStar {...newProps} />);
    expect(component.find('MaterialDesignIcon').props().style).toEqual(sampleStyle);
  });

  describe('Sample props rendering', () => {
    const component = shallow(<ChainItemStar {...sampleProps}/>);
    it('renders icon component ', () => {
      expect(component.find('MaterialDesignIcon').props().name).toBe('star');
    });
    it('assigns correct color', () => {
      const sampleStyle = {
        color: colors[sampleProps.color][500],
      };
      expect(component.find('MaterialDesignIcon').props().style).toEqual(sampleStyle);
    });
  });
});
