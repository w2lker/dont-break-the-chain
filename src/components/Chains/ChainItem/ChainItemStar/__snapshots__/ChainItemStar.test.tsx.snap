// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chain Item Score matches snapshot 1`] = `
"<ChainItemStar color=\\"green\\" value={6}>
  <MaterialDesignIcon name=\\"star\\" style={{...}}>
    <span className=\\"material-icon mdi mdi-star\\" style={{...}} />
  </MaterialDesignIcon>
</ChainItemStar>"
`;
